packages/df-blogs/src/EditComment.tsx

```diff
@@ -1 +1,233 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Button, Message } from 'semantic-ui-react';
+import { Form, Field, withFormik, FormikProps } from 'formik';
+import * as Yup from 'yup';
+import { History } from 'history';
+
+import TxButton from '@polkadot/joy-utils/TxButton';
+import { SubmittableResult } from '@polkadot/api';
+import { withCalls, withMulti } from '@polkadot/ui-api/with';
+
+import * as JoyForms from '@polkadot/joy-utils/forms';
+import { AccountId, Text, Bool } from '@polkadot/types';
+import { Option, Vector } from '@polkadot/types/codec';
+import { PostId, Post, PostData, PostUpdate, BlogId } from './types';
+import Section from '@polkadot/joy-utils/Section';
+import { useMyAccount } from '@polkadot/joy-utils/MyAccountContext';
+import { queryBlogsToProp, UrlHasIdProps } from './utils';
+import { CommentId, Comment, CommentUpdate, CommentData } from './types';
+
+const buildSchema = (p: ValidationProps) => Yup.object().shape({
+  body: Yup.string()
+    // .min(p.minTextLen, `Your post is too short. Minimum length is ${p.minTextLen} chars.`)
+    // .max(p.maxTextLen, `Your post description is too long. Maximum length is ${p.maxTextLen} chars.`)
+    .required('Comment body is required')
+});
+
+type ValidationProps = {
+  // minTitleLen: number,
+  // maxTitleLen: number,
+  // minTextLen: number,
+  // maxTextLen: number
+};
+
+type OuterProps = ValidationProps & {
+  history?: History,
+  id?: PostId,
+  commentId?: CommentId,
+  struct?: Comment
+};
+
+type FormValues = CommentData & {
+  body: string
+};
+
+type FormProps = OuterProps & FormikProps<FormValues>;
+
+const LabelledField = JoyForms.LabelledField<FormValues>();
+
+const InnerForm = (props: FormProps) => {
+  const {
+    history,
+    id,
+    commentId,
+    struct,
+    values,
+    dirty,
+    isValid,
+    isSubmitting,
+    setSubmitting,
+    resetForm
+  } = props;
+
+  const {
+    body
+  } = values;
+
+  const onSubmit = (sendTx: () => void) => {
+    if (isValid) sendTx();
+  };
+
+  const onTxCancelled = () => {
+    setSubmitting(false);
+  };
+
+  const onTxFailed = (_txResult: SubmittableResult) => {
+    setSubmitting(false);
+  };
+
+  const onTxSuccess = (_txResult: SubmittableResult) => {
+    setSubmitting(false);
+
+    // TODO get id of newly created post and redirect.
+    // goToView(id);
+  };
+
+  const isNew = struct === undefined;
+
+  const buildTxParams = () => {
+    if (!isValid) return [];
+
+    const json = JSON.stringify(
```

Comment:
Just put in on one line because it's quite short.
```suggestion
    const json = JSON.stringify({ body });
```